File: Data Structures & Algorithms/three-integer-sum/submission-0.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        
        Arrays.sort(nums);
        int length = nums.length;
        List<List<Integer>> resultList = new ArrayList<>();
        for( int i = 0 ; i < length-2 ; i++ ){
            if( i > 0 && nums[i] == nums[i-1]) continue;
            int left = i+1;
            int right = length-1;
            while( left < right ) {
                int sum = nums[i] + nums[left] + nums[right];
                if( sum == 0 ) {
                    List<Integer> combinationList = new ArrayList<>();
                    combinationList.add(nums[i]);
                    combinationList.add(nums[left]);
                    combinationList.add(nums[right]);
                    resultList.add(combinationList);
                    left++;
                    while( nums[left-1] == nums[left] && left < right ) left++;
                    right--;
                    while( nums[right] == nums[right+1] && left < right ) right--;
                }
                else if( sum < 0 ) left++;
                else right--;
            
            }
        }  
        return resultList; 
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
class Solution:
    def search( self ,nums: List[int],target: int) -> int:
        left = 0
        right = len(nums)-1
        while left < right :
            mid = left + ( right - left )//2
            if  nums[mid] == target:
                return mid
            elif nums[left] < nums[right] and nums[mid] < nums[right]:
                if target > nums[mid]:
                    left = mid+1
                else:
                    right = mid
            elif nums[mid] < nums[right]:
                if target > nums[right] or target < nums[mid]:
                    right = mid
                else:
                    left = mid+1        
            else:
                if target <= nums[right] or target > nums[mid]:
                    left = mid+1
                else:
                    right = mid
        if nums[left] == target:
            return left
        return -1 
    
    
    
    